src/services/schema-form.js

```diff
@@ -409,22 +593,33 @@ angular.module('schemaForm').provider('schemaForm',
       globalOptions = globalOptions || {};
 
       if (stripNullType(schema.type) === 'object') {
-        angular.forEach(schema.properties, function(v, k) {
+        var modifiedProperties = combinePropertyDependencies(schema);
+
+        angular.forEach(modifiedProperties, function(v, k) {
           if (ignore[k] !== true) {
             var required = schema.required && schema.required.indexOf(k) !== -1;
             var def = defaultFormDefinition(k, v, {
-              path: [k],         // Path to this property in bracket notation.
-              lookup: lookup,    // Extra map to register with. Optimization for merger.
-              ignore: ignore,    // The ignore list of paths (sans root level name)
-              required: required, // Is it required? (v4 json schema style)
+              path: [k],            // Path to this property in bracket notation.
+              lookup: lookup,       // Extra map to register with. Optimization for merger.
+              ignore: ignore,       // The ignore list of paths (sans root level name)
+              required: required,   // Is it required? (v4 json schema style)
               global: globalOptions // Global options, including form defaults
             });
+
             if (def) {
               form.push(def);
             }
           }
         });
 
+        if (schema.required && Array === schema.required.constructor) {
```

Comment:
Pushes required objects to the top, and then sorts.
This is where the non-required objects can be removed for the initial rollout once all is done.